firefox-ios/Client/Frontend/Browser/Search/SearchViewController.swift

```diff
@@ -454,6 +462,15 @@ class SearchViewController: SiteTableViewController,
                   let suggestion = suggestions[safe: indexPath.row],
                   let url = defaultEngine.searchURLForQuery(suggestion)
             else { return }
+            */
+            guard let defaultEngine = viewModel.searchEnginesManager?.defaultEngine else { return }
+
+            searchTelemetry?.selectedResult = .searchSuggest
+            // Assume that only the default search engine can provide search suggestions.
+            guard let suggestions = viewModel.suggestions,
+                  indexPath.row < min(suggestions.count, 4) else { return }
+            let suggestion = suggestions[indexPath.row]
```

Comment:
I guess this are the main lines that change, maybe we could be more specific on what to comment out with Ecosia and change to make future upgrades even more manageable?